sources/Renderer/Metal/MTRenderContext.mm

```diff
@@ -102,7 +108,25 @@
 
 bool MTRenderContext::OnSetVideoMode(const VideoModeDescriptor& videoModeDesc)
 {
-    return true; // do nothing
+    // There is only one scale factor. Verify that it works on both axes.
+    CGFloat scale = videoModeDesc.resolution.width / view_.bounds.size.width;
```

Comment:
I'm not experienced enough with the MacOS view/view-controller or `CAMetalLayer` system (or whatever type `view_.layer` is) to provide a sophisticated review on this function, but it seems to do its job, so looks good to me. The only request I have for this one is that I prefer to be more explicit about type casts:
```cpp
uint32_t scaledHeight = static_cast<uint32_t>(view_.bounds.size.height * scale + 0.5);
```